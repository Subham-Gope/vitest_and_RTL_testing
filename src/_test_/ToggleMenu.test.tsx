// import { describe, test, expect } from "vitest";
// import { screen, render } from "@testing-library/react";
// import userEvent from "@testing-library/user-event";
// import ToggleMenu from "../components/ToggleMenu";

// describe("Testing the ToggleMenu.tsx file", () => {
//   test("toggles menu on pressing Shift + m", async () => {
//     // Setting up userEvent
//     const user = userEvent.setup();

//     const { container } = render(
//       <ToggleMenu
//         message={"Testing Component"}
//         assignmentNumber={5}
//         status={true}
//       />
//     );

//     expect(container).toMatchSnapshot();

//     const menu = screen.getByRole("navigation", { hidden: true });
//     expect(menu).toBeInTheDocument();
//     expect(menu).toHaveClass("translate-x-full");
//     expect(menu).toHaveClass("opacity-0");

//     // initiating user action
//     await user.keyboard("[AltLeft>][KeyM][/ShiftLeft]");
//     expect(menu).toHaveClass("translate-x-0");
//     expect(menu).toHaveClass("opacity-100");
//   });
// });
