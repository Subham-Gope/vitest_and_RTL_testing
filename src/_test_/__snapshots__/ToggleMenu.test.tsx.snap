// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Testing the ToggleMenu.tsx file > toggles menu on pressing Shift + m 1`] = `
<div>
  <div
    class=" flex flex-col overflow-x-hidden relative min-h-screen"
  >
    <h1>
      Message: 
      Testing Component
      , Assigment Number: 
      5
      , Completion Status:
      , Toggle State: 
      false
    </h1>
    <div
      class="h-screen w-[200px] bg-gray-100 flex  flex-col items-center mt-4 absolute top-[30px] right-0 translate-x-full  opacity-0  transition-all duration-300 ease-in-out delay-100"
      role="navigation"
    >
      <ul
        class="flex flex-col gap-4 px-8 py-12 list-none decoration-0"
      >
        <li>
          Home
        </li>
        <li>
          About
        </li>
        <li>
          Contact
        </li>
      </ul>
    </div>
  </div>
</div>
`;
